spec/app_spec.rb

```diff
@@ -134,7 +134,7 @@
     stdin = StringIO.new("")
     app = App.new(stdout, stdin)
 
-    allow(stdin).to receive(:gets).and_return(
+    expect(stdin).to receive(:gets).and_return(
```

Comment:
This works as a failing then passing test, since rspec implicitly expects it to be called twice. I would expect something a bit more explicit in the test. The test description could change. You could do something after the application was previously quitting and make assertions that output was correct.

It does beg the question too about whether all allows on gets should be expects. There might be other easy to see expectation fails in there.